css/css-display/display-contents-accessibility.html

```diff
@@ -0,0 +1,321 @@
+        <thead>
+          <tr style="display: contents;" data-expectedrole="row" data-testname="tr in table with display:contents has row role" class="ex">
+            <th style="display: contents;" data-expectedrole="columnheader" data-testname="td within tr both with display:contents, in table with display:contents, has columnheader role" class="ex">x</th>
+            <th>x</th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr>
+            <td style="display: contents;" data-expectedrole="cell" data-testname="td within tr in table, all with display:contents, has cell role" class="ex">x</td>
+            <td>x</td>
+          </tr>
+        </tbody>
+      </table>
+
+    <div role="table" style="display: contents;" data-expectedrole="table" data-testname="div with role table with display:contents has table role" class="ex">
+        <div role="rowgroup" style="display: contents;" data-expectedrole="rowgroup" data-testname="div with role rowgroup in div with role table, both with display:contents, has rowgroup role" class="ex">
+            <div role="row">
+                <div role="columnheader" style="display: contents;" data-expectedrole="columnheader" data-testname="div with role columnheader with display:contents, in div with role rowgroup within div with table role both with display:contents, has columnheader role">x</div>
+                <div role="columnheader">x</div>
+            </div>
+        </div>
+        <div role="rowgroup">
+            <div role="row">
+                <div role="cell"></div>
+                <div role="cell" style="display: contents;" data-expectedrole="cell" data-testname="div with cell role and display:contents, within div with role row contained in a div with role table both with display:contents, has cell role" class="ex"></div>
+            </div>
+        </div>
+    </div>
+
+    <table role="grid" style="display: contents;" data-expectedrole="grid" data-testname="table with role grid and display:contents has grid role">
+        <thead>
+          <tr style="display: contents;" data-expectedrole="row" data-testname="tr within table with role grid, both with display:contents, has row role">
+            <th style="display: contents;" data-expectedrole="columnheader" data-testname="th within table with role grid, both with display:contents, has columnheader role">x</th>
+            <th>x</th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr>
+            <td style="display: contents;" data-expectedrole="cell" data-testname="td within table with role grid, both with display:contents, has cell role" class="ex">x</td>
+            <td>x</td>
+          </tr>
+        </tbody>
+      </table>
+
+      <div role="grid" style="display: contents;" data-expectedrole="grid" data-testname="div with role grid with display:contents has grid role" class="ex">
```

Comment:
Chromium/WebKit pass, but Gecko fails with `expected "grid" but got "table"`… Hmmm.

Are there different results if you make this grid focusable with `tabindex="0"`? If so, your example might be triggering some other reasonable implementation detail heuristic in Gecko.

Reply:
@cookiecrook I've added a separate test `role="grid"` with `tabindex="0"`: https://github.com/web-platform-tests/wpt/pull/43740/commits/1c94eebfc62a23b0d8c9236aea473945a32df692.

It looks like Gecko still fails: https://wpt.fyi/results/css/css-display/display-contents-accessibility.html?diff&filter=ADC&run_id=5091859742851072&run_id=5100840251031552.